sdk/dapp-kit/src/hooks/wallet/useSignAndExecuteTransactionBlock.ts

```diff
@@ -39,43 +40,71 @@ type UseSignAndExecuteTransactionBlockMutationOptions = Omit<
 		unknown
 	>,
 	'mutationFn'
->;
+> & {
+	executeFromWallet?: boolean;
+};
 
 /**
  * Mutation hook for prompting the user to sign and execute a transaction block.
  */
 export function useSignAndExecuteTransactionBlock({
 	mutationKey,
+	executeFromWallet,
```

Comment:
I thought signing and executing all from the wallet was the standard way of doing things (I might be confused bc it's late though lol), should we rename this to `executeFromApp` and default it to `false`?

Reply:
I think it has been the default way of doing things, but it's less efficient

Reply:
Yeah it's how it's been done, and how other wallets worked before. To me the big downsides is that it's harder to debug and error handle (as you have two actions coupled into one), and is slower for the transaction to show up in your fullnode (since it's executing against the wallet, then has to sync out to wherever you're querying from, vs being executed in the same place you query). Historically folks have also been confused when after awaiting this call, they still don't see the transaction in their fullnode (which is why we introduced `waitForTransactionBlock`).